source/classes.tex

```diff
@@ -2488,18 +2486,52 @@
 \nontermdef{conversion-declarator}\br
     ptr-operator \opt{conversion-declarator}
 \end{bnf}
-shall have no non-object parameters and
-specifies a conversion from \tcode{X} to
+
+\pnum
+A declaration
+whose \grammarterm{declarator-id} has
+an \grammarterm{unqualified-id} that is a \grammarterm{conversion-function-id}
+declares a \defnadj{conversion}{function};
+its \grammarterm{declarator} shall be
+a function declarator\iref{dcl.fct} of the form
+\begin{ncsimplebnf}
+ptr-declarator \terminal{(} parameter-declaration-clause \terminal{)} \opt{cv-qualifier-seq}\br
+\bnfindent \opt{ref-qualifier-seq} \opt{noexcept-specifier} \opt{attribute-specifier-seq}
+\end{ncsimplebnf}
+where the \grammarterm{ptr-declarator} consists solely of
+an \grammarterm{id-expression},
+an optional \grammarterm{attribute-specifier-seq}, and
+optional surrounding parentheses, and
+the \grammarterm{id-expression} has one of the following forms:
+\begin{itemize}
+\item
+in a \grammarterm{member-declaration} that belongs to
+the \grammarterm{member-specification} of a class or class template
+but is not a friend declaration\iref{class.friend},
+the \grammarterm{id-expression} is a \grammarterm{conversion-function-id};
+\item
+otherwise, the \grammarterm{id-expression} is a \grammarterm{qualified-id}
+whose \grammarterm{unqualified-id} is a \grammarterm{conversion-function-id}.
+\end{itemize}
+
+\pnum
+A conversion function  shall have no non-object parameters and
```

Comment:
The core issue doesn't include "non-object".

I suppose you added it to account for explicit object parameters.

Is that really enough? No need to restrict it to an explicit object parameter, but any amount of non-object parameters is OK? Is there wording elsewhere that make these cases ill-formed, and thus saying "non-object" is enough?

How about the function type specified below? Does it need a revision, too?

Reply:
I think the paper misrepresents the status quo? So tihs seems like a "plausible merge".

Reply:
A conversion function with an explicit this parameter seems plausible, but there must not be other parameters. Further cleanup (if any) should be in a separate editorial or core issue.